packages/tx/src/types.ts

```diff
@@ -99,6 +117,25 @@ export function isAccessList(input: AccessListBytes | AccessList): input is Acce
   return !isAccessListBytes(input) // This is exactly the same method, except the output is negated.
 }
 
+export function isAuthorizationListBytes(
+  input: AuthorizationListBytes | AuthorizationList
+): input is AuthorizationListBytes {
+  if (input.length === 0) {
+    return true
+  }
+  const firstItem = input[0]
+  if (Array.isArray(firstItem)) {
+    return true
+  }
+  return false
+}
+
+export function isAuthorizationList(
```

Comment:
i think typescript should be able to infer the other type by itself if you use !isAuthorizationList

but am not 100% sure of typescript shenanigans

Reply:
I have copied this pattern from the access lists :)